pkg/operator/resource/resourceapply/generic.go

```diff
@@ -388,3 +405,17 @@ func (c *ClientHolder) secretsGetter() corev1client.SecretsGetter {
 	}
 	return v1helpers.CachedSecretGetter(c.kubeClient.CoreV1(), c.kubeInformers)
 }
+
+func (c *ClientHolder) deploymentsGetter() appv1client.DeploymentsGetter {
+	if c.kubeClient == nil {
+		return nil
+	}
+	return c.kubeClient.AppsV1()
+}
+
+func (c *ClientHolder) daemonSetsGetter() appv1client.DaemonSetsGetter {
+	if c.kubeClient == nil {
+		return nil
+	}
+	return c.kubeClient.AppsV1()
+}
```

Comment:
You can likely delete these two methods. They seem to be modeled after `configMapsGetter()` and `secretsGetter()` above, but those functions are designed to return a cached client (preferably), whereas the new ones here always return a non-cached client.

Reply:
fixed